app/build.gradle

```diff
@@ -33,9 +34,19 @@ android {
 
 dependencies {
 
+    // Import the BoM for the Firebase platform
+    implementation platform('com.google.firebase:firebase-bom:29.1.0')
+
     implementation 'androidx.appcompat:appcompat:1.4.1'
     implementation 'com.google.android.material:material:1.5.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
+    implementation 'com.google.firebase:firebase-auth:21.0.1'
```

Comment:
Since we're using the Firebase BoM, I think you can safely delete this line (we would still have the version-less firebase-auth dependency below)
```suggestion
```